core/commands/p2p.go

```diff
@@ -196,6 +202,40 @@ Example:
 	},
 }
 
+// checkPort checks whether target multiaddr contains tcp or udp protocol
+// and whether the port is equal to 0
+func checkPort(target ma.Multiaddr) error {
+	// get tcp or udp port from multiaddr
+	getPort := func() (string, error) {
+		sport, _ := target.ValueForProtocol(ma.P_TCP)
+		if sport != "" {
+			return sport, nil
+		}
+
+		sport, _ = target.ValueForProtocol(ma.P_UDP)
+		if sport != "" {
+			return sport, nil
+		}
+		return "", fmt.Errorf("address does not contain tcp or udp protocol")
```

Comment:
We'll want to support unix sockets in the future, but until that happens this can probably stay here